packages/@ember/-internals/metal/lib/array.ts

```diff
@@ -16,16 +19,23 @@ export function objectAt<T>(array: T[] | EmberArray<T>, index: number): T | unde
   }
 }
 
+// Ideally, we'd use MutableArray.detect but for unknown reasons this causes
+// the node tests to fail strangely.
+function isMutableArray<T>(obj: unknown): obj is MutableArray<T> {
+  return obj != null && typeof (obj as MutableArray<T>).replace === 'function';
+}
+
 export function replace<T>(
-  array: NativeArray<T> | MutableArray<T>,
+  array: T[] | MutableArray<T>,
   start: number,
   deleteCount: number,
   items: T[] = EMPTY_ARRAY as []
 ): void {
-  if (Array.isArray(array)) {
-    replaceInNativeArray(array, start, deleteCount, items);
-  } else {
+  if (isMutableArray(array)) {
```

Comment:
`Array.isArray` will match as true for `Ember.NativeArray` but since that has a special `replace` method we want to use that instead.